src/posix/platform/backbone.cpp

```diff
@@ -42,7 +42,7 @@
 
 char         gBackboneNetifName[IFNAMSIZ] = "";
 unsigned int gBackboneNetifIndex          = 0;
-#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_MULTICAST_ROUTING_ENABLE
+#if OPENTHREAD_POSIX_CONFIG_BACKBONE_ROUTER_MULTICAST_ROUTING_ENABLE
```

Comment:
Should we include in the name something more specific to MFC / MRT? In the future, do we envision supporting multiple different ways of supporting multicast routing on posix platform?

Reply:
Currently only linux is supported for posix multicast routing, the implementation is guarded by `#if __linux__`. Other posix systems may support MFCs differently(e.g. netBSD), or not support MFCs. Some parts of the MulticastRoutingManager could possibly be reused by different implementations, e.g entry of SetUp(), TearDown(), HandleStateChange().

I think we could keep this flag general for posix systems and when adding support for other posix systems use the OS macros to guard the implementations. Thoughts?